stripe/src/main/java/com/stripe/android/IssuingCardPinService.java

```diff
@@ -0,0 +1,290 @@
+package com.stripe.android;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.annotation.VisibleForTesting;
+
+import com.stripe.android.exception.APIConnectionException;
+import com.stripe.android.exception.APIException;
+import com.stripe.android.exception.AuthenticationException;
+import com.stripe.android.exception.CardException;
+import com.stripe.android.exception.InvalidRequestException;
+
+import org.json.JSONException;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+/*
+ * Methods for retrieval / update of a Stripe Issuing card
+ */
+public class IssuingCardPinService
+        implements EphemeralKeyManager.KeyManagerListener<IssuingCardEphemeralKey> {
```

Comment:
this is fine because it's the way we've been doing it, but I slightly prefer having an instance of `EphemeralKeyManager.KeyManagerListener<IssuingCardEphemeralKey>` within this class rather than implementing it on the class level so that we don't add `onKeyUpdate` and `onKeyError` to the public API of the class

Reply:
ah ok, noted for next time 😬